pkg/registry/entry.go

```diff
@@ -129,8 +129,41 @@ func (u *urlEntry) Contains(e Entry) bool {
 	return false
 }
 
-func (u *urlEntry) Match(s string) bool {
-	return u.Contains(ParseEntry(s))
+func (u *urlEntry) Match(s string) (result bool) {
+	q := ParseEntry(s)
+	query, ok := q.(URLEntry)
+	if !ok {
+		return false
+	}
+
+	result = u.Contains(query)
+	if result {
+		return
+	}
+
+	if query.URL().Scheme != "" {
```

Comment:
currently collapsing this nested if into a && instead, just did this at first cause it was a little easier to see the logic I was implementing